ui-v2/app/utils/search/filterable.js

```diff
@@ -0,0 +1,38 @@
+import RSVP, { Promise } from 'rsvp';
+export default function(EventTarget = RSVP.EventTarget, P = Promise) {
+  // TODO: Class-ify
+  return function(filter) {
+    return EventTarget.mixin({
+      value: '',
+      add: function(data) {
+        this.data = data;
+        return this;
+      },
+      search: function(term = '') {
+        this.value = term === null ? '' : term.trim();
+        // specifically no return here we return `this` instead
+        // right now filtering is sync but we introduce an async
+        // flow now for later on
+        P.resolve(
+          this.value !== ''
+            ? this.data.filter(item => {
+                return filter(item, { s: term });
+              })
+            : this.data
+        ).then(data => {
```

Comment:
Right now this is synchronous, but I've used this approach so I know I can either think of implementing asynchronous searching either using the backend, or putting the search in a web worker.